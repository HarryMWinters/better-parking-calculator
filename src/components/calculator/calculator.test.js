import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
configure({ adapter: new Adapter() });

import Calculator from "./calculator";

const parkingInfo = {
  "Economy Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 9.0,
    "Per Week": 63.0
  },
  "Short-Term Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 24.0,
    "Per Week": 168.0
  },
  "Long-Term Surface Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 12.0,
    "Per Week": 84.0
  },
  "Long-Term Garage Parking": {
    "Per Hour": 2.0,
    "Daily Maximum": 17.0,
    "Per Week": 119.0
  },
  "Valet Parking": {
    "Per Hour": null,
    // Valet parking is only charged per day.
    "Daily Maximum": 24.0,
    "Per Week": 169.0
  }
};

let testCalculator = shallow(<Calculator parkingInfo={parkingInfo} />);

it("Renders correctly when parking type is NOT selected", () => {
  const entryDate = new Date("2019-08-02T22:16:43.383Z");
  const exitDate = new Date("2019-09-02T22:16:43.383Z");
  testCalculator.setState({ entryDateTime: entryDate, exitDateTime: exitDate });
  const tree = renderer.create(testCalculator).toJSON();
  expect(tree).toMatchSnapshot();
});

// it("Renders correctly when parking type IS selected", () => {
//   const entryDate = new Date("2019-08-02T22:16:43.383Z");
//   const exitDate = new Date("2019-09-02T22:17:43.383Z");
//   testCalculator.setState({
//     entryDateTime: entryDate,
//     exitDateTime: exitDate,
//     parkingOptionSelected: "Long-Term Garage Parking"
//   });
//   const tree = renderer.create(testCalculator).toJSON();
//   expect(tree).toMatchSnapshot();
// });
