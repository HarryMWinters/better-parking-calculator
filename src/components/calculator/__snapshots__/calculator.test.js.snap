// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when parking type IS selected 1`] = `
<div
  className="calculator"
>
  <div
    className="Container"
  >
    <h4>
      Parking Type 
    </h4>
    <div
      className="dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-secondary"
        disabled={false}
        id="dropdown-basic"
        onClick={[Function]}
        type="button"
      >
        Long-Term Garage Parking
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Entered at:
          </label>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="2019-08-02"
                name="Entered at:"
                onBlur={[Function]}
                type="date"
              />
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="15:16"
                onBlur={[Function]}
                type="time"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="col"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Exited at: 
          </label>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="2019-09-02"
                name="Exited at: "
                onBlur={[Function]}
                type="date"
              />
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="15:17"
                onBlur={[Function]}
                type="time"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <button
    className="btn btn-primary btn-block btn-lg"
    disabled={false}
    id="calcButton"
    onClick={[Function]}
    type="button"
  >
    Calculate!
  </button>
  <div>
    <div
      className="container"
      style={
        Object {
          "marginTop": "1em",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <div
            className="card"
          >
            <div
              className="card-header"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Total Time:
            </div>
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                --:--
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <div
            className="card"
          >
            <div
              className="card-header"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Price: 
            </div>
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                 
                --
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly when parking type is NOT selected 1`] = `
<div
  className="calculator"
>
  <div
    className="Container"
  >
    <h4>
      Parking Type 
    </h4>
    <div
      className="dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-secondary"
        disabled={false}
        id="dropdown-basic"
        onClick={[Function]}
        type="button"
      >
        Choose Parking Option
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Entered at:
          </label>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="2019-08-02"
                name="Entered at:"
                onBlur={[Function]}
                type="date"
              />
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="15:16"
                onBlur={[Function]}
                type="time"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      className="col"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Exited at: 
          </label>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="2019-09-02"
                name="Exited at: "
                onBlur={[Function]}
                type="date"
              />
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                defaultValue="15:16"
                onBlur={[Function]}
                type="time"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <button
    className="btn btn-primary btn-block btn-lg"
    disabled={false}
    id="calcButton"
    onClick={[Function]}
    type="button"
  >
    Calculate!
  </button>
  <div>
    <div
      className="container"
      style={
        Object {
          "marginTop": "1em",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <div
            className="card"
          >
            <div
              className="card-header"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Total Time:
            </div>
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                --:--
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <div
            className="card"
          >
            <div
              className="card-header"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Price: 
            </div>
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                 
                --
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
